usr.sbin/ppp/iface.c

```diff
@@ -201,64 +201,42 @@ iface_Create(const char *name)
 static int
 iface_addr_Zap(const char *name, struct iface_addr *addr, int s)
 {
-  struct ifaliasreq ifra;
+  struct ifreq ifr;
 #ifndef NOINET6
-  struct in6_aliasreq ifra6;
+  struct in6_ifreq ifr6;
 #endif
-  struct sockaddr_in *me4, *msk4, *peer4;
-  struct sockaddr_storage ssme, sspeer, ssmsk;
+  struct sockaddr_in *me4;
+  struct sockaddr_storage ssme;
   int res, saved_errno;
 
-  ncprange_getsa(&addr->ifa, &ssme, &ssmsk);
-  ncpaddr_getsa(&addr->peer, &sspeer);
+  ncprange_getsa(&addr->ifa, &ssme, NULL);
   res = 0;
 
   switch (ncprange_family(&addr->ifa)) {
   case AF_INET:
-    memset(&ifra, '\0', sizeof ifra);
-    strncpy(ifra.ifra_name, name, sizeof ifra.ifra_name - 1);
-
-    me4 = (struct sockaddr_in *)&ifra.ifra_addr;
+    memset(&ifr, '\0', sizeof ifr);
+    strncpy(ifr.ifr_name, name, sizeof ifr.ifr_name - 1);
```

Comment:
Does the code pay any attention to the ifr_name or can we just use a designated initializer to set ifr_addr?

Reply:
I cannot see any code in the kernel that uses `ifr_name`. It is copied in `ifioctl()` (`sys/net/if.c`), and I think it might be better to initialise it in case someone comes up with an idea to print `ifr_name` for debugging purposes for example.